src/librustc_mir/interpret/memory.rs

```diff
@@ -791,6 +791,28 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         self.get_raw(ptr.alloc_id)?.read_c_str(self, ptr)
     }
 
+    /// Reads a 0x0000-terminated u16-sequence from memory. Returns them as a Vec<u16>.
+    /// Terminator 0x0000 is not included in the returned Vec<u16>.
+    ///
+    /// Performs appropriate bounds checks.
+    pub fn read_wide_str(&self, ptr: Scalar<M::PointerTag>) -> InterpResult<'tcx, Vec<u16>> {
+        let mut ptr = self.force_ptr(ptr)?; // We need to read at least 1 byte, so we *need* a ptr.
+        let allocation = self.get_raw(ptr.alloc_id)?;
+        let mut u16_seq = Vec::new();
+
+        loop {
+            let single_u16 =
+                allocation.read_scalar(&self, ptr, self.data_layout().pointer_size)?.to_u16()?;
```

Comment:
The size should be "2" as you are reading two bytes.
And the first parameter should just be `self`, not `&self` -- `self` already is a reference; `&self` is of type `&&Memory` and that is one reference too much.